packages/es-components/src/components/containers/popoverTest/Popup.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { injectGlobal } from 'styled-components';
+
+import popoverStyles from './popoverStyles';
+
+import { Manager, Target, Popper, Arrow } from 'react-popper';
+import Transition from 'react-transition-group/Transition';
+import RootCloseWrapper from 'react-overlays/lib/RootCloseWrapper';
+
+const PopperContainer = styled.div`
+  display: ${props => (props.transitionState === 'exited' ? 'none' : 'block')};
+`;
+
+const defaultStyle = {
+  transition: 'opacity 0.2s ease-in-out',
+  opacity: 0,
+  zIndex: 5
+};
+const transitionStyles = {
+  entering: { opacity: 0 },
+  entered: { opacity: 1 }
+};
+
+function getArrowSize(size) {
+  switch (size) {
+    case 'sm':
+      return '5';
+    case 'lg':
+      return '20';
+    case 'none':
+      return '0';
+    default:
+      return '10';
+  }
+}
+
+function Popup({
+  name,
+  transitionIn,
+  transitionTimeout,
+  placement,
+  children,
+  trigger,
+  onHide,
+  arrowSize,
+  popperModifiers,
+  disableArrow,
+  disableRootClose,
+  hasTitle,
+  theme
+}) {
+  const arrowStyles = popoverStyles(
+    name,
+    theme.colors,
+    getArrowSize(arrowSize),
+    hasTitle
+  );
+  /* eslint-disable no-unused-expressions */
+  injectGlobal`${arrowStyles}`;
```

Comment:
It is agreed this is heavy handed, but we do not have a whole lot of choice. `react-popper` only offers classes and attributes to style with. We've tried requiring downstream projects to consume a css file in order to use this project but that turned out to be very unfriendly. While we go against `styled-components`'s suggestion in using `injectGlobal` at least we're providing the necessary css only when we need it.

Reply:
Alex is correct. Due to how `react-popper` relies on it's data attribute tags for styling both the popovers and arrows, there is no real way to do this without the `injectGlobal`. I've tried to figure out other ways because it's not something I particularly like either, but there's been no luck.

Reply:
So it is cool that `injectGlobals` to an extent does prevent duplicates from being injected. Pulling down your branch changing the name of the sample popover adds a new entry of css to head, but changing it back does not. This shouldn't be a concern for production use, as the name of a single instance of the new popover should not be changing constantly, but it is something to keep an eye on.